frontend/src/metabase/admin/datamodel/containers/MetricApp.jsx

```diff
@@ -6,73 +6,86 @@ import MetabaseAnalytics from "metabase/lib/analytics";
 
 import MetricForm from "./MetricForm.jsx";
 
-import { metricEditSelectors } from "../selectors";
 import * as actions from "../datamodel";
-import { clearRequestState } from "metabase/redux/requests";
-import { fetchTableMetadata } from "metabase/redux/metadata";
 import { getMetadata } from "metabase/selectors/metadata";
+import Metrics from "metabase/entities/metrics";
+import Tables from "metabase/entities/tables";
 
 const mapDispatchToProps = {
   ...actions,
-  fetchTableMetadata,
-  clearRequestState,
+  createMetric: Metrics.actions.create,
   onChangeLocation: push,
 };
 
 const mapStateToProps = (state, props) => ({
-  ...metricEditSelectors(state, props),
   metadata: getMetadata(state, props),
 });
 
-@connect(
-  mapStateToProps,
-  mapDispatchToProps,
-)
-export default class MetricApp extends Component {
-  async componentWillMount() {
-    const { params, location } = this.props;
+@Metrics.load({
+  id: (state, props) => parseInt(props.params.id),
+  wrapped: true,
+})
+@Tables.load({ id: (state, props) => props.metric.table_id, wrapped: true })
```

Comment:
Is this double object loader ok? It seems to work, but I probably can't rely on `props.metric.table_id` being set before `Table.load` calls the id function.

Reply:
In general stacking object loaders is ok. However, I believe the requests aren't parallelized (hence why `props.metric` is never null), so try to avoid doing it too much.